code-challenges.test.js

```diff
@@ -22,41 +22,135 @@ const numbersArray2 = [24, 27, 30, 33, 36]
+// Tests passed successfully and the function ran as expected.
+
 
 
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
+// Psuedocode:
+// 1. Create a function called arrayCapitalized
+// 2. Pass in a parameter called array. This will be used for arrays that are passed into the function.
+// 3. Return statement
+// 4. Use .map function on the parameter, because the exact same number of values will be returned, but only the first letter of each word will be changed.
+    // 4a. Use value & arrow function to iterate through each index of the array
+    // 4b. Use the charAt(0).toUpperCase() function to make each string index of 0 uppercase
+    // 4c. Use value.substring(1) to concatenate the rest of the word
+
 // a) Create a test with expect statements for each of the variables provided.
 
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+// // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
+// // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
+describe("arrayCapitalized", () => {
+    const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+    const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
+
+  it("takes in an array of words and returns an array with all the words capitalized.", () => {
+    expect(arrayCapitalized(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+    expect(arrayCapitalized(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+  })
+})
+
+// //     ReferenceError: arrayCapitalized is not defined
+
+
+// // b) Create the function that makes the test pass.
+
+const arrayCapitalized = (array) => {
+    return array.map(value => value.charAt(0).toUpperCase() + value.substring(1))
+}
+
+// Outcome:
+// arrayCapitalized
+// ✓ takes in an array of words and returns an array with all the words capitalized. (2 ms)
+// Tests passed successfully and the function ran as expected.
 
-// b) Create the function that makes the test pass.
 
 
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
-// a) Create a test with expect statements for each of the variables provided.
+// Psuedocode:
+
+// I tried using higher order functions, using the .filter method, but gave me an error: TypeError: firstVowel.filter is not a function. I realized that .map/.filter can only be used in arrays. The question is asking about using a string input, so I changed my method to a basic for/if condition.
```

Comment:
In the future, leave that code in here and just comment it out if it doesn't work.  I like to see the process, and perhaps I can help guide you to making it work.